### ISSUE TITLE
Batch multiple images

### ISSUE THREAD

**Issue by frinyvonnick** [frinyvonnick/node-html-to-image#2]
I would be nice to create multiple images using the same template in only one instance of puppeteer

**Comment by frinyvonnick** [comment]
I think about giving ability to the user to give a browser instance to avoid creating a new instance each time node-html-to-image is called.

**Comment by robinmetral** [comment]
This would be great! I needed to generate 2k images and had to make my laptop open and close 2k headless chrome instances recursively 🙈

**Comment by frinyvonnick** [comment]
Just had an idea of implementation. Here is how it would be used:

```js
// Default usage
const nodeHtmlToImage = require('node-html-to-image')

nodeHtmlToImage({
  output: './image.png',
  html: '<html><body>Hello world!</body></html>'
})
  .then(() => console.log('The image was created successfully!'))

// Batch usage
const imageBatcher = nodeHtmlToImage.initializeImageBatcher()

const contents = [{ name: 'Pierre' }, { name: 'Paul' }, { name: 'Jacques' }]
Promise.all(contents.map(content => {
  imageBatcher.batch({
    output: `./image-${content.name}.png`,
    content,
    html: '<html><body>Hello {{name}}!</body></html>'
  })
}))
  .then(() => imageBatcher.close())
```

What do you think of it @robinmetral?

**Comment by robinmetral** [comment]
Looks good! I'm not sure how the implementation would look like (not very familiar with Puppeteer), but I'm thinking that it could be abstracted away to keep the API simple:

```javascript
// single image
nodeHtmlToImage({
  output: './image.png',
  html: '<html><body>Hello {{name}}!</body></html>',
  content: { name: 'world' }
})
  .then(() => console.log('The image was created successfully!'))

// multiple images
nodeHtmlToImage({
  output: './image.png',
  html: '<html><body>Hello {{name}}!</body></html>',
  content: [{ name: 'Pierre' }, { name: 'Paul' }, { name: 'Jacques' }]
})
  .then(() => console.log('The images were created successfully!'))
```

This way, generating multiple images would be trivial for users, they just need to pass an array to `content` instead of a single object.

Of course you would still need something like your `initializeImageBatcher()` under the hood.

Would this work?

**Comment by frinyvonnick** [comment]
It looks great @robinmetral. I'll try something like this 👍

**Comment by robinmetral** [comment]
Sounds good! I'm a bit too busy these days to tackle this unfortunately, but I'm happy to review or test it!

**Comment by frinyvonnick** [comment]
Hi @robinmetral 👋 

I opened a pull request (#38) with the first implementation of this feature. You propose your help to test or review it. It would help me a lot if you could do it 🙏

**Comment by frinyvonnick** [comment]
This has been fixed and published in [3.0.0](https://github.com/frinyvonnick/node-html-to-image/releases/tag/v3.0.0) 🚀
